tests/unit/components/SharedComponents/Buttons/Button.test.js

```diff
@@ -0,0 +1,121 @@
+      // Snapshot test
+      expect( screen ).toMatchSnapshot();
+    } );
+
+    it( "has no accessibility errors", async () => {
+      const button = <Button text="NEUTRAL BUTTON" />;
+      // Snapshot test
+      expect( button ).toBeAccessible();
+    } );
+  } );
+
+  describe( "focus", () => {
+    it( "renders correctly", async () => {
+      render( <Button level="focus" text="FOCUS BUTTON" /> );
+      // Snapshot test
+      expect( screen ).toMatchSnapshot();
+    } );
+
+    it( "has no accessibility errors", async () => {
+      const button = <Button level="focus" text="FOCUS BUTTON" />;
+      // Snapshot test
+      expect( button ).toBeAccessible();
+    } );
+  } );
+
+  describe( "warning", () => {
+    it( "renders correctly", async () => {
+      render( <Button level="warning" text="WARNING BUTTON" /> );
+      // Snapshot test
+      expect( screen ).toMatchSnapshot();
+    } );
+
+    it( "has no accessibility errors", async () => {
+      const button = <Button level="warning" text="WARNING BUTTON" />;
+      // Snapshot test
+      expect( button ).toBeAccessible();
+    } );
+  } );
+
+  describe( "primary disabled", () => {
+    it( "renders correctly", async () => {
+      render( <Button level="primary" text="PRIMARY DISABLED" disabled /> );
+      // Snapshot test
+      expect( screen ).toMatchSnapshot();
+    } );
+
+    it( "has no accessibility errors", async () => {
+      const button = (
+        <Button level="primary" text="PRIMARY DISABLED" disabled />
+      );
+      // Snapshot test
+      expect( button ).toBeAccessible();
+    } );
+  } );
+
+  describe( "neutral disabled", () => {
+    it( "renders correctly", async () => {
+      render( <Button text="NEUTRAL DISABLED" disabled /> );
+      // Snapshot test
+      expect( screen ).toMatchSnapshot();
+    } );
+
+    it( "has no accessibility errors", async () => {
+      const button = <Button text="NEUTRAL DISABLED" disabled />;
+      // Snapshot test
+      expect( button ).toBeAccessible();
+    } );
+  } );
+
+  describe( "focus disabled", () => {
+    it( "renders correctly", async () => {
+      render( <Button level="focus" text="FOCUS DISABLED" disabled /> );
+      // Snapshot test
+      expect( screen ).toMatchSnapshot();
+    } );
+
+    it( "has no accessibility errors", async () => {
+      const button = <Button level="focus" text="FOCUS DISABLED" disabled />;
+      // Snapshot test
+      expect( button ).toBeAccessible();
+    } );
+  } );
+
+  describe( "warning disabled", () => {
+    it( "renders correctly", async () => {
+      render( <Button level="warning" text="WARNING DISABLED" disabled /> );
+      // Snapshot test
+      expect( screen ).toMatchSnapshot();
+    } );
+
+    it( "has no accessibility errors", async () => {
+      const button = (
+        <Button level="warning" text="WARNING DISABLED" disabled />
+      );
+      // Snapshot test
+      expect( button ).toBeAccessible();
+    } );
+  } );
+} );
```

Comment:
Lots of boilerplate here. Any disadvantage to dynamically making the tests? I was thinking something like:

```javascript
["primary", "netural"].forEach( level => {
  describe( level, ( ) => {
    it( "renders correctly", async () => {
      render( <Button level={level} text={`${level.toUpperCase( )} BUTTON`} /> );
      expect( screen ).toMatchSnapshot();
    } );

    it( "has no accessibility errors", async () => {
      const button = <Button level={level} text={`${level.toUpperCase( )} BUTTON`} />;
      expect( button ).toBeAccessible();
    } );
  } );
} );
```

Maybe a bit less readable, but less to maintain.

Reply:
Have added a loop with `describe.each`.